src/Codeception/Util/Debug.php

```diff
@@ -38,6 +40,28 @@ public static function isEnabled(): bool
         return (bool)self::$output;
     }
 
+    public static function pause(array $vars = []): void
+    {
+        if (!self::isEnabled()) {
+            return;
+        }
+
+        $pauseShell = new PauseShell();
+        $psy = $pauseShell->getShell();
+        $psy->setScopeVariables($vars);
+        $backtrace = debug_backtrace(DEBUG_BACKTRACE_PROVIDE_OBJECT, 3);
+        if (isset($backtrace[1]['object'])) {
+            // set the scope of test class
+            $pauseShell->addMessage('Use $this-> to access current object');
+            $psy->setBoundObject($backtrace[1]['object']);
+        } elseif (isset($backtrace[2]['object'])) {
```

Comment:
It seems we are guessing here, maybe we could check if object implements `\PHPUnit\Framework\Test` too.
Actor should be one step closer, we could detect it by looking for anan instance of `\Codeception\Actor`.
It could help to avoid code duplication too.

Reply:
I agree on this, thanks. Could you try this?
I am not sure I understand this code good enough to handle these cases the best way

Reply:
I will try to improve it.

Reply:
@Naktibalda I thought a bit for this and no, we can't use checks like you propose:

`codecept_pause` must be available from any place in the code, not only from tests.